src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2815,29 +2838,30 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                             );
                             throw SemanticAbort();
                         }
-                        if (!ASRUtils::check_equal_type(ASRUtils::expr_type(val), value_type)) {
-                            std::string vtype = ASRUtils::type_to_str_python(ASRUtils::expr_type(val));
+                        if (!ASRUtils::check_equal_type(ASRUtils::expr_type(tmp_value), value_type)) {
+                            std::string vtype = ASRUtils::type_to_str_python(ASRUtils::expr_type(tmp_value));
                             std::string totype = ASRUtils::type_to_str_python(value_type);
                             diag.add(diag::Diagnostic(
                                 "Type mismatch in dictionary value, the types must be compatible",
                                 diag::Level::Error, diag::Stage::Semantic, {
                                     diag::Label("type mismatch (found: '" + vtype + "', expected: '" + totype + "')",
-                                            {val->base.loc})
+                                            {tmp_value->base.loc})
                                 })
                             );
                             throw SemanticAbort();
                         }
                         ASR::expr_t* se = ASR::down_cast<ASR::expr_t>(
                                 ASR::make_Var_t(al, x.base.base.loc, s));
-                        tmp = make_DictInsert_t(al, x.base.base.loc, se, key, val);
-                        return;
+                        tmp = make_DictInsert_t(al, x.base.base.loc, se, key, tmp_value);
+                        tmp_vec.push_back(tmp);
```

Comment:
Something like this?
```suggestion
                        tmp = nullptr;
                        tmp_vec.push_back(make_DictInsert_t(al, x.base.base.loc, se, key, tmp_value));
```

Reply:
Wait, or is this just one? In that case, just put it into `tmp`, and leave `tmp_vec` be.

Reply:
No, we can have many of them. We have a `continue` statement below this.